Cluster data source

breml opened the pull request:
Replaces #279 and #284.

Example (mainly useful for a clustered Incus setup):

```hcl
terraform {
  required_providers {
    incus = {
      source = "lxc/incus"
    }
  }
}

provider "incus" {}

data "incus_cluster" "this" {
  remote = "cluster"
  lifecycle {
    postcondition {
      condition     = alltrue(self.is_clustered ? [for i, v in self.members : v.status == "Online"] : [])
      error_message = "All servers must be online."
    }
  }
}
```

Resolves: #268

breml commented:
Go 1.25 has been released, looks like this breaks golangci-lint for now.

stgraber commented:
As discussed earlier today, we'll want to remove the fallback logic, so if a system isn't clustered, the member list should be empty.

Once that's done, we should be good to go. We can ignore the golangci-lint stuff for now, hopefully they'll fix their binaries soon now that 1.25 is out.

breml commented:
> As discussed earlier today, we'll want to remove the fallback logic, so if a system isn't clustered, the member list should be empty.

@stgraber I added the commits to remove the fallback logic for the non-clustered case and updated the tests and the documentation accordingly.
